orc8r/cloud/go/sqorc/error_checker.go

```diff
@@ -44,8 +44,8 @@ func GetErrorChecker() ErrorChecker {
 
 func (c SQLiteErrorChecker) GetError(err error) error {
 	if e, ok := err.(sqlite3.Error); ok {
-		switch e.Code {
-		case sqlite3.ErrConstraint:
+		switch e.ExtendedCode {
+		case sqlite3.ErrConstraintUnique:
```

Comment:
Would it make sense to treat a primary key violation (`sqlite3.ErrConstraintPrimaryKey`) the same way?

Reply:
for sure, I just didn't want to add granularity that's not currently required by the code. There are a great deal of error codes available both in sqlite3 and postgres and I figured we should either cover them all or just the ones that are actually used in the code. I picked the latter approach, but when the need arises the above switch should definitely be extended.

Reply:
I agree, this could be improved by adding more error codes in the future but I wouldn't do that now.

I just think a primary key error would be triggered by the same cause - trying to enter something that already exists. So it could return the same error.

Reply:
ah, gotcha, I can add it to the switch